datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -701,6 +701,29 @@ macro_rules! make_try_abs_function {
     }};
 }
 
+// Decimal types should keep the same precision and scale by using `with_data_type()`.
+// https://github.com/apache/arrow-rs/issues/4644
+macro_rules! make_decimal_abs_function {
+    ($ARRAY_TYPE:ident) => {{
+        |args: &[ArrayRef]| {
+            let array = downcast_arg!(&args[0], "abs arg", $ARRAY_TYPE);
+            let res: $ARRAY_TYPE = array
+                .try_unary(|x| {
+                    // Valid Decimal values should never overflow
+                    x.checked_abs().ok_or_else(|| {
+                        ArrowError::InvalidArgumentError(format!(
+                            "Invalid {} value {} when calculating abs",
+                            stringify!($ARRAY_TYPE),
+                            x
```

Comment:
```suggestion
                            "{} value {} overflows when calculating abs",
                            stringify!($ARRAY_TYPE),
                            x
```

Reply:
@viirya I would like to emphasize that the input is illegal for the same reason mentioned above.

Reply:
Looks like this is unnecessary if we use `wrapping_abs`.